### ISSUE TITLE
Expose list of common dart source dirs as targets for formatter, linter, analyzer, etc.

### ISSUE THREAD

**Issue by seaneagan** [google/grinder.dart#239]
It'd be nice to default `Analyzer`, `Dartfmt` (others?) to running on all common dart source directories.  If we had [this DEP](https://github.com/lrhn/dep-optionals) then we could just make the paths optional in `Analyzer.analyze` and `Dartfmt.format/dryRun`, but maybe more generally useful would be exposing the list of dirs:

```dart
final List<Directory> sourceDirs = ['bin', 'example', 'lib', 'test', 'tool', 'web'].map((path) => new Directory(path)).toList();
```

or `dartSourceDirs`?

`web` can contain dart source code, right?

**Comment by zoechi** [comment]
Yes, `web` usually contains `dart` source. 
I haven't tried if analyzer or formatter choke on not existing directories.
`sourceDirs.where((d) => d.existsSync())` isn't too hard, maybe adding directly?
Having such a list is handy, I usually reuse it in several tasks.

**Comment by seaneagan** [comment]
I like it!

```
final List<Directory> sourceDirs = ['bin', 'example', 'lib', 'test', 'tool', 'web'].map((path) => new Directory(path)).where((d) => d.existsSync()).toList();
```

**Comment by devoncarew** [comment]
Hmm, maybe a static list of top-level dirs that can contain dart source, and a method that will return the subset that actually exist for the current project? So,

```dart
const List<String> sourceDirs;
List<String> getExistingSourceDirs();
```

**Comment by seaneagan** [comment]
sgtm, though I'd make both getters and use `Directory` instead of `String`:

```dart
const List<Directory> sourceDirs;
List<Directory> get existingSourceDirs => sourceDirs.where((d) => d.existsSync()).toList();
```

**Comment by seaneagan** [comment]
or probably more performant:

```dart
List<Directory> get existingSourceDirs => Directory.current.listSync().where(sourceDirs.contains).toList();
```

**Comment by devoncarew** [comment]
I generally don't think of getters as things that could throw exceptions.

**Comment by seaneagan** [comment]
The style guide doesn't mention that as a criteria:

https://www.dartlang.org/articles/style-guide/#do-use-a-getter-for-operations-that-conceptually-access-a-property

is it a Java thing?  I don't personally see a need for that distinction.

**Comment by zoechi** [comment]
Fixed by #260
